lib/config-generator.js

```diff
@@ -463,6 +467,7 @@ class ConfigGenerator {
             // avoid CORS concerns trying to load things like fonts from the dev server
             headers: { 'Access-Control-Allow-Origin': '*' },
             // required by FriendlyErrorsWebpackPlugin
+            hot: true,
```

Comment:
What about requiring the user to pass the `--hot` flag in at the command line:

```
./node_modules/.bin/encore dev-server --hot
```

Then, we would detect this (similar to how we're intercepting a few options already in the `encore` executable, like `--https`) and use it above to conditionally add `HotModuleReplacementPlugin`. Otherwise, we're forcing the `dev-server` to *only* be used with `hot` mode (I'm not sure if there's an issue with that, but forcing it feels a little weird)